### PR TITLE
e2e incognito mode

### LINKED ISSUE #7722: add a script to run e2e tests with Cypress in incognito mode
add a small script so we can open the test in incognito mode with one line of code. That might be useful in the future if we want to test without any cache issues.

### PR DESCRIPTION
**Pull request by n-germain-work**
Fix #7722 

# Possible improvements

## plugins/index.ts files

All `index` files have the following code 👍
```
module.exports = (on, config) => {
  on('before:browser:launch', (browser, launchOptions) => {
    if (browser.name === 'chrome' && config.env.incognito) {
      launchOptions.args.push('--incognito');
    }
    return launchOptions;
  });
};
```
It would be best to have a global approach to avoid duplicated code.

### CODE REVIEW

File: apps/festival/festival-e2e/src/plugins/index.ts [hunk 1]
```diff
@@ -20,9 +20,14 @@ export default (async (on, config) => {
   console.log('Node version:', process.version);
   console.log('Config: ', config);
   // A plugin example. Can modify printed configs.
-  on('before:browser:launch', (arg1, arg2) => {
-    console.log('Plugin Arg1', arg1);
-    console.log('Plugin Arg2', arg2);
+
+  // works by calling : nx run festival-e2e:e2e:emulator-incognito
+  // see angular.json at the root of the project
+  on("before:browser:launch", (browser, launchOptions) => {
```
Comment (phpgeekfr) [inline apps/festival/festival-e2e/src/plugins/index.ts:26]: always use single quotes `'`

File: apps/festival/festival-e2e/src/plugins/index.ts [hunk 2]
```diff
@@ -20,9 +20,14 @@ export default (async (on, config) => {
...
+    if (browser.name === "chrome" && config.env.incognito) {
```
Comment (phpgeekfr) [inline apps/festival/festival-e2e/src/plugins/index.ts:27]: same

File: apps/festival/festival-e2e/src/plugins/index.ts [hunk 3]
```diff
@@ -20,9 +20,14 @@ export default (async (on, config) => {
   console.log('Node version:', process.version);
   console.log('Config: ', config);
   // A plugin example. Can modify printed configs.
-  on('before:browser:launch', (arg1, arg2) => {
-    console.log('Plugin Arg1', arg1);
-    console.log('Plugin Arg2', arg2);
+
+  // works by calling : nx run festival-e2e:e2e:emulator-incognito
+  // see angular.json at the root of the project
```
Comment (phpgeekfr) [inline apps/festival/festival-e2e/src/plugins/index.ts:25]: maybe you can add a command in `package.json` in the  `scripts` section? for the 4 apps ?

File: apps/festival/festival-e2e/src/plugins/index.ts [hunk 4]
```diff
@@ -20,9 +20,14 @@ export default (async (on, config) => {
   console.log('Node version:', process.version);
   console.log('Config: ', config);
   // A plugin example. Can modify printed configs.
-  on('before:browser:launch', (arg1, arg2) => {
-    console.log('Plugin Arg1', arg1);
-    console.log('Plugin Arg2', arg2);
+
+  // works by calling : nx run festival-e2e:e2e:emulator-incognito
```
Comment (phpgeekfr) [inline apps/festival/festival-e2e/src/plugins/index.ts:24]: so this is the equivalent of `npx cross-env nx e2e festival-e2e --configuration=emulator, incognito` ?

File: package.json [hunk 5]
```diff
@@ -30,7 +30,11 @@
     "lint:backend-functions": "cross-env nx run backend-functions:lint --configuration=$NG_CONFIG",
     "e2e": "cross-env nx run-many --target=e2e --all  --configuration=$NG_CONFIG",
     "e2e:catalog": "cross-env nx e2e catalog-e2e --configuration=$NG_CONFIG --headless",
+    "e2e:catalog:incognito": "cross-env nx e2e catalog-e2e --configuration=emulator-incognito",
+    "e2e:catalog:incognito:watch": "cross-env nx e2e catalog-e2e --configuration=emulator-incognito --watch",
     "e2e:festival": "cross-env nx e2e festival-e2e --configuration=$NG_CONFIG --headless",
+    "e2e:festival:incognito": "cross-env nx e2e festival-e2e --configuration=emulator-incognito",
+    "e2e:festival:incognito:watch": "cross-env nx e2e festival-e2e --configuration=emulator-incognito --watch",
```
Comment (phpgeekfr) [inline package.json:37]: don't we want `cross-env nx e2e festival-e2e --configuration=emulator-incognito -- --watch` ?

Comment (phpgeekfr) [reply package.json:37]: can you add command for other apps as well ?

File: apps/financiers/financiers-e2e/src/plugins/index.js [hunk 6]
```diff
@@ -12,6 +12,10 @@
 // the project's config changing)
 
 module.exports = (on, config) => {
-  // `on` is used to hook into various events Cypress emits
-  // `config` is the resolved Cypress config
+  on('before:browser:launch', (browser, launchOptions) => {
+    if (browser.name === 'chrome' && config.env.incognito) {
+      launchOptions.args.push("--incognito");
```
Comment (phpgeekfr) [inline apps/financiers/financiers-e2e/src/plugins/index.js:17]: use single quotes

File: package.json (hunk 5 above)
Comment (n-germain-work) [reply package.json:37]: `-- --watch` is if we launched with `npm`. The command above works fine.

For the other apps, i'll add them when needed 👌

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**